pubsub/gochannel/fanin.go

```diff
@@ -0,0 +1,139 @@
+package gochannel
```

Comment:
As long as it's not specific for a gochannel, I think that it may be worth to move to https://github.com/ThreeDotsLabs/watermill/tree/master/components :wink: